sql/schema_changer_test.go

```diff
@@ -667,4 +668,42 @@ CREATE UNIQUE INDEX vidx ON t.test (v);
 	if eCount != count {
 		t.Fatalf("read the wrong number of rows: e = %d, v = %d", eCount, count)
 	}
+
+	// Verify that a table delete in the middle of a backfill works properly.
+	// The backfill will terminate in the middle, and the delete will
+	// successfully delete all the table data.
+	//
+	// This test could be made its own test but is placed here to speed up the
+	// testing.
+
+	backfillNotification = make(chan bool)
+	// Run the schema change in a separate goroutine.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		// Start schema change that eventually runs a backfill.
+		_, err := sqlDB.Exec("CREATE UNIQUE INDEX bar ON t.test (v)")
+		if !testutils.IsError(err, "descriptor ID .* not found") {
```

Comment:
but I thought the whole point is that the sql statement doesn't return an error. And this internal error looks like it definitely shouldn't be exposed.
So I'd swallow it in all the callers of `getDescriptorNotDeleted`. Which means you'd need a different mechanism to figure out that you've actually tested the "backfill encouters deleted table" code path. See my comment below about using KV "command filters" - you could use them to assert that some keys have never been backfilled, even though the `CREATE INDEX` statement finished.